### ISSUE TITLE
Influxdb metric output not grouped by backend/server

### ISSUE THREAD

**Issue by dbienapfl-dataminr** [containous/traefik#3146]
### Do you want to request a *feature* or report a *bug*?
Bug

### What did you do?
Installed v1.6.4rc4 and turned on the influx metrics reporting

### What did you expect to see?
Metrics properly divided by backend/backend server

### What did you see instead?
Metrics grouped by backend that included _all_ backend hosts and not just the hosts/urls associated with that backend.  

Traefik is publishing on the 10 second interval - however - it is publishing data for _all_ hosts not just the hosts associated with the reported/tagged backend.  Looking at the data - it appears to cycle through all backend hosts over time.

<details>
<summary>Raw data from influx:</summary>

```
> select * from "traefik.entrypoint.requests.total" WHERE "environment" = 'Test' limit 200
name: traefik.entrypoint.requests.total
time                Cluster          Role    Team agent    az         backend                   code collectiontype count entrypoint environment host                      method protocol url
----                -------          ----    ---- -----    --         -------                   ---- -------------- ----- ---------- ----------- ----                      ------ -------- ---
1523048652804381015 ops-traefik-Test traefik ops  telegraf us-east-1b primespeed_load_generator 200  application    18    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.31:80
1523048662804337838 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    54    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.58:5556
1523048672804352766 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    33    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.51.52:80
1523048682804303043 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    34    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.31:80
1523048692804328428 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    53    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.34.92:5556
1523048702804290241 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    63    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.51.52:80
1523048712804290875 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    70    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.31:80
1523048722804273462 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    83    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.50.222:8081
1523048732804352153 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    65    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.51.52:80
1523048742804315104 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    58    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.31:80
1523048752804264041 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    75    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.34.92:5556
1523048762804314674 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    52    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.51.52:80
1523048772804341681 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    74    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.31:80
1523048782804244093 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    59    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.58.94:8081
1523048792804356653 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    60    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.51.52:80
1523048802804311266 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    53    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.31:80
1523048812804365088 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    66    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.58.94:8081
1523048822804345775 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    59    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.51.52:80
1523048832804319448 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    59    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.31:80
1523048842804296719 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    87    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.58.94:8081
1523048852804354296 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    54    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.51.52:80
1523048862804404858 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    43    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.31:80
1523048872804315095 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    65    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.58.94:8081
1523048882804370672 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    51    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.51.52:80
1523048892804333487 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    61    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.31:80
1523048902804336626 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    58    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.50.12:8888
1523048912804363902 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    55    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.51.52:80
1523048922810333089 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    41    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.31:80
1523048932804359360 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    61    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.34.92:5556
1523048942808725742 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    63    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.51.52:80
1523048952804251785 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    75    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.31:80
1523048962804295187 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    67    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.58:5556
1523048972804310889 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    53    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.51.52:80
1523048982804286466 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    62    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.31:80
1523048992804356320 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    89    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.50.12:8888
1523049002804326366 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    44    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.51.52:80
1523049012804290215 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    99    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.31:80
1523049022804275382 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    74    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.58.94:8081
1523049032804270629 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    38    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.51.52:80
1523049042804331835 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    58    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.31:80
1523049052804321188 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    75    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.58.94:8081
1523049062804345771 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    52    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.51.52:80
1523049072804259874 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    76    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.31:80
1523049082804303541 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    64    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.34.92:5556
1523049092804333183 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    36    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.51.52:80
1523049102804282481 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    24    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.31:80
1523049112804267528 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    25    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.58.94:8081
1523049122804318223 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    14    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.51.52:80
1523049132804347212 ops-traefik-Test traefik ops  telegraf us-east-1b translateproxy            200  application    27    https      Test        ops-traefik-test-7.dm.vpc POST   http     http://172.19.59.31:80
```

</details>

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
bash-4.4# /traefik version
Version:      v1.6.0-rc4
Codename:     tetedemoine
Go version:   go1.10.1
Built:        2018-04-04_01:40:48PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

<details>
<summary>Configuration</summary>

```toml
# bash-4.4# cat /config/traefik.toml
################################################################
# Global configuration
################################################################

# Enable debug mode
#
# Optional
# Default: false
#
debug = false

# Log level
#
# Optional
# Default: "ERROR"
#
logLevel = "ERROR"

# Entrypoints to be used by frontends that do not specify any entrypoint.
# Each frontend can specify its own entrypoints.
#
# Optional
# Default: ["http"]
#
# defaultEntryPoints = ["http", "https"]
defaultEntryPoints = ["https"]

# Entrypoints definition
#
# Optional
# Default:

[entryPoints]
    [entryPoints.traefik]
      address = ":8080"
    [entryPoints.https]
      address = ":443"
    [entryPoints.https.tls]
      [[entryPoints.https.tls.certificates]]
      certFile = "/ssl_config/service.crt"
      keyFile = "/ssl_config/service.key"

[accessLog]

# Sets the file path for the access log. If not specified, stdout will be used.
# Intermediate directories are created if necessary.
#
# Optional
# Default: os.Stdout
#
# filePath = "/path/to/log/log.txt"

# Format is either "json" or "common".
#
# Optional
# Default: "common"
#
format = "json"

[metrics]
  [metrics.influxdb]

    # InfluxDB's address.
    #
    # Required
    # Default: "localhost:8089"
    #
    address = "localhost:8087"

    # InfluxDB push interval
    #
    # Optional
    # Default: "10s"
    #
    pushinterval = "10s"

[api]
  entryPoint = "traefik"
  dashboard = true

################################################################
# DynamoDB configuration backend
################################################################

# Enable DynamoDB configuration backend.
[dynamodb]

# Region to use when connecting to AWS.
#
# Required
#
region = "us-east-1"

# DyanmoDB Table Name.
#
# Optional
# Default: "traefik"
#
tableName = "traefik_routing_Dev"

# Enable watch DynamoDB changes.
#
# Optional
# Default: true
#
watch = true

# Polling interval (in seconds).
#
# Optional
# Default: 15
#
refreshSeconds = 10
```

</details>

<details>
<summary> backends at the time of the above data collection</summary>

```json
{
  "dynamodb":{
    "backends":{
      "pipeline_api":{
        "servers":{
          "data-pipeline-api-Test-1":{
            "url":"http://172.19.34.92:5556",
            "weight":0
          },
          "data-pipeline-api-Test-2":{
            "url":"http://172.19.59.58:5556",
            "weight":0
          }
        },
        "loadBalancer":{
          "method":"wrr",
          "stickiness":{
            "cookieName":"3983224698189305043"
          }
        },
        "healthCheck":{
          "path":"/health_check",
          "port":5556,
          "interval":"10"
        }
      },
      "primespeed_load_generator":{
        "servers":{
          "ops-primespeed-load-generator-Test-10":{
            "url":"http://172.19.51.52:80",
            "weight":0
          },
          "ops-primespeed-load-generator-Test-9":{
            "url":"http://172.19.59.31:80",
            "weight":0
          }
        },
        "loadBalancer":{
          "method":"wrr",
          "stickiness":{
            "cookieName":"4981722866654283343"
          }
        },
        "healthCheck":{
          "path":"/",
          "port":80,
          "interval":"10"
        }
      },
      "schema_registry":{
        "servers":{
          "data-schema-registry-Test-1":{
            "url":"http://172.19.50.12:8888",
            "weight":0
          },
          "data-schema-registry-Test-2":{
            "url":"http://172.19.42.25:8888",
            "weight":0
          }
        },
        "loadBalancer":{
          "method":"wrr",
          "stickiness":{
            "cookieName":"-7539791700055208120"
          }
        },
        "healthCheck":{
          "path":"/health_check",
          "port":8888,
          "interval":"10"
        }
      },
      "translateproxy":{
        "servers":{
          "data-translateproxy-Test-48":{
            "url":"http://172.19.50.222:8081",
            "weight":0
          },
          "data-translateproxy-Test-49":{
            "url":"http://172.19.58.94:8081",
            "weight":0
          }
        },
        "loadBalancer":{
          "method":"wrr",
          "stickiness":{
            "cookieName":"2775863960964999634"
          }
        },
        "healthCheck":{
          "path":"/status.json",
          "port":8081,
          "interval":"10"
        }
      }
    },
    "frontends":{
      "pipeline_api":{
        "entryPoints":[
          "https"
        ],
        "backend":"pipeline_api",
        "routes":{
          "pipeline_api":{
            "rule":"Headers: X-Dm-Role, pipeline_api"
          }
        },
        "passHostHeader":true,
        "priority":0,
        "basicAuth":null,
        "headers":{

        }
      },
      "primespeed_load_generator":{
        "entryPoints":[
          "https"
        ],
        "backend":"primespeed_load_generator",
        "routes":{
          "primespeed_load_generator":{
            "rule":"Headers: X-Dm-Role, primespeed_load_generator"
          }
        },
        "passHostHeader":true,
        "priority":0,
        "basicAuth":null,
        "headers":{

        }
      },
      "schema_registry":{
        "entryPoints":[
          "https"
        ],
        "backend":"schema_registry",
        "routes":{
          "schema_registry":{
            "rule":"Headers: X-Dm-Role, schema_registry"
          }
        },
        "passHostHeader":true,
        "priority":0,
        "basicAuth":null,
        "headers":{

        }
      },
      "translateproxy":{
        "entryPoints":[
          "https"
        ],
        "backend":"translateproxy",
        "routes":{
          "translateproxy":{
            "rule":"Headers: X-Dm-Role, translateproxy"
          }
        },
        "passHostHeader":true,
        "priority":0,
        "basicAuth":null,
        "headers":{

        }
      }
    }
  }
}
```
</details>

**Comment by Juliens** [comment]
WDYT @marco-jantke ?

**Comment by mmatur** [comment]
Hi @dbienapfl-dataminr 

Thanks for your interest on Træfik. 

This issue is due a bug in go-kit, a [PR](https://github.com/go-kit/kit/pull/404) has already been done to fix it.

I am working on a PR on Træfik to update go-kit dependency.

**Comment by traefiker** [comment]
Closed by #3189.
